posthog/models/constance.py

```diff
@@ -0,0 +1,46 @@
+import json
+from contextlib import contextmanager
+from typing import Any
+
+from django.db import models
+
+from posthog.settings import CONSTANCE_CONFIG, CONSTANCE_DATABASE_PREFIX
+
+
+class Constance(models.Model):
+    class Meta:
+        constraints = [models.UniqueConstraint(fields=["key"], name="unique key",)]
+
+    key: models.CharField = models.CharField(max_length=128, null=False, blank=False)
+    raw_value: models.CharField = models.CharField(max_length=1024, null=False, blank=True)
+
+    @property
+    def value(self):
+        return json.loads(self.raw_value)
+
+
+def get_dynamic_setting(key: str) -> Any:
```

Comment:
q: does constance itself always hit the database? does it not keep anything in memory?

Reply:
Yes it did - you can see this in tests even.